### PR TITLE
Fix f-string calls

### PR DESCRIPTION
**Pull request by drewrisinger**

### CODE REVIEW

File: qiskit/circuit/library/generalized_gates/pauli.py [hunk 1]
```diff
@@ -71,9 +71,9 @@ def validate_parameter(self, parameter):
                 return parameter
             else:
                 raise CircuitError(
-                    "Parameter string {0} should contain only " "'I', 'X', 'Y', 'Z' characters"
+                    "Parameter string {0} should contain only 'I', 'X', 'Y', 'Z' characters"
                 )
         else:
             raise CircuitError(
-                "Parameter {0} should be a string of " "'I', 'X', 'Y', 'Z' characters"
+                "Parameter {0} should be a string of 'I', 'X', 'Y', 'Z' characters"
```
Comment (jakelishman) [inline qiskit/circuit/library/generalized_gates/pauli.py:78]: These two strings look like they're missing formatting calls - perhaps they could both become
```python
    f"Parameter {parameter} should be a string of 'I', 'X', 'Y' and 'Z' characters."
```

File: qiskit/extensions/hamiltonian_gate.py [hunk 2]
```diff
@@ -123,7 +123,7 @@ def validate_parameter(self, parameter):
             return float(parameter)
         else:
             raise CircuitError(
-                "invalid param type {} for gate  " "{}".format(type(parameter), self.name)
+                "invalid param type {} for gate  {}".format(type(parameter), self.name)
```
Comment (jakelishman) [inline qiskit/extensions/hamiltonian_gate.py:126]:
```suggestion
                "invalid param type {} for gate {}".format(type(parameter), self.name)
```

File: qiskit/extensions/quantum_initializer/initializer.py [hunk 3]
```diff
@@ -362,7 +362,7 @@ def validate_parameter(self, parameter):
             return complex(parameter.item())
         else:
             raise CircuitError(
-                "invalid param type {} for instruction  " "{}".format(type(parameter), self.name)
+                "invalid param type {} for instruction  {}".format(type(parameter), self.name)
```
Comment (jakelishman) [inline qiskit/extensions/quantum_initializer/initializer.py:365]:
```suggestion
                "invalid param type {} for instruction {}".format(type(parameter), self.name)
```

File: qiskit/extensions/quantum_initializer/isometry.py [hunk 4]
```diff
@@ -299,7 +299,7 @@ def validate_parameter(self, parameter):
             return parameter
         else:
             raise CircuitError(
-                "invalid param type {} for gate  " "{}".format(type(parameter), self.name)
+                "invalid param type {} for gate  {}".format(type(parameter), self.name)
```
Comment (jakelishman) [inline qiskit/extensions/quantum_initializer/isometry.py:302]:
```suggestion
                "invalid param type {} for gate {}".format(type(parameter), self.name)
```

File: qiskit/pulse/schedule.py [hunk 5]
```diff
@@ -1779,7 +1779,7 @@ def _check_nonnegative_timeslot(timeslots: TimeSlots):
         if chan_timeslots:
             if chan_timeslots[0][0] < 0:
                 raise PulseError(
-                    "An instruction on {} has a negative " " starting time.".format(chan)
+                    "An instruction on {} has a negative  starting time.".format(chan)
```
Comment (jakelishman) [inline qiskit/pulse/schedule.py:1782]:
```suggestion
                    "An instruction on {} has a negative starting time.".format(chan)
```

File: qiskit/qasm/node/format.py [hunk 6]
```diff
@@ -37,7 +37,7 @@ def qasm(self, prec=None):
         """Return the corresponding format string."""
         if prec is not None:
             warnings.warn(
-                "Parameter 'Format.qasm(..., prec)' is no longer used and is being " "deprecated.",
+                "Parameter 'Format.qasm(..., prec)' is no longer used and is being deprecated.",
```
Comment (jakelishman) [inline qiskit/qasm/node/format.py:40]: All of these strings in the QASM 2 module (and possibly some other modules in this PR) were already removed in #7208 - you probably just want to fix the merge conflicts by throwing out these changes.

Alternatively, if you did this PR by `sed` or similar, then maybe just rebase onto `main` and redo it - might be quicker!

**Review by jakelishman — COMMENTED** [review]
Thanks for looking at this!  A lot of the merge conflicts will be because old deprecation warnings were removed in #7208 - you may need to rebase this PR, and drop changes in places that have been deleted on our side.

Obviously nothing problematic was introduced by this PR, but while checking it I spotted a couple of places with questionable spacing and a couple of places where the format strings had gone wrong - would you mind just poking those into shape while we're at it?

**Comment by drewrisinger** [comment]
@jakelishman thanks for the review. Looks like most of it was resolved by using an up-to-date version of the ``main`` branch. 🤦 Updated the title accordingly.

**Review by jakelishman — APPROVED** [review]
These changes look good to me - thanks for checking all the others before, and that's quite bad luck that recent deprecation tidy-ups took almost all of them out!

**Pull request merged**